overlord/hookstate/hooks.go

```diff
@@ -70,6 +71,22 @@ func SetupPreRefreshHook(st *state.State, snapName string) *state.Task {
 	return task
 }
 
+func SetupAutoRefreshGatingHook(st *state.State, snapName string, base, restart bool) *state.Task {
+	hookSup := &HookSetup{
+		Snap:     snapName,
+		Hook:     "gate-auto-refresh",
+		Optional: true,
+		IgnoreError: true,
```

Comment:
I don't think we want to ignore errors, we want to install a handler generator that provides handler that will manage what to do on errors?